### DISCUSSION TITLE
Stored procedure

### DISCUSSION THREAD

**Discussion by vanillajonathan** [PRQL/prql#3004] [Ideas]
Maybe PRQL could support executing stored procedures?

Maybe there could be a "execute" keyword in PRQL so that:

```elm
execute SelectAllCustomers
```
Would transform into the SQL:
```sql
EXEC SelectAllCustomers
```

**Comment by max-sixty** [comment]
Because PRQL only issues one SQL query per PRQL query, and we target analytical use cases — this probably isn't high on the list.

Happy to leave this open to collect ideas though!

**Reply by vanillajonathan** [reply]
The stored procedure could return a value that could be used in a analytical use case.
```elm
let foo = execute usp_get_foo
from customers
filter status == foo
